Failure to compile on Debian Sid

```
[ 63%] Building CXX object src/CMakeFiles/minetest.dir/gettext.cpp.o
In file included from /home/james/Games/Minetest/minetest/src/gettext.cpp:24:0:
/usr/include/libintl.h:39:14: error: expected unqualified-id before ‘char’
 extern char *gettext (const char *__msgid)
              ^
/usr/include/libintl.h:39:14: error: expected initializer before ‘char’
src/CMakeFiles/minetest.dir/build.make:3038: recipe for target 'src/CMakeFiles/minetest.dir/gettext.cpp.o' failed
make[2]: *** [src/CMakeFiles/minetest.dir/gettext.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:218: recipe for target 'src/CMakeFiles/minetest.dir/all' failed
make[1]: *** [src/CMakeFiles/minetest.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

Reverting b11720af459d44a553cb5d23ef776a632fb30a65 results in no build error, however.

```
[ 29%] Building CXX object src/CMakeFiles/minetestserver.dir/gettext.cpp.o
In file included from /home/james/Games/Minetest/minetest/src/gettext.cpp:24:0:
/home/james/Games/Minetest/minetest/src/gettext.h:28:3: error: #error "should not be reached"
  #error "should not be reached"
   ^~~~~
[ 30%] Building CXX object src/CMakeFiles/minetestserver.dir/httpfetch.cpp.o
/home/james/Games/Minetest/minetest/src/gettext.h: In function ‘const wchar_t* wgettext(const char*)’:
/home/james/Games/Minetest/minetest/src/gettext.h:44:35: error: ‘gettext’ was not declared in this scope
  return utf8_to_wide_c(gettext(str));
                                   ^
/home/james/Games/Minetest/minetest/src/gettext.h: In function ‘std::__cxx11::string strgettext(const string&)’:
/home/james/Games/Minetest/minetest/src/gettext.h:49:29: error: ‘gettext’ was not declared in this scope
  return gettext(text.c_str());
                             ^
[ 30%] Building CXX object src/CMakeFiles/minetestserver.dir/inventory.cpp.o
src/CMakeFiles/minetestserver.dir/build.make:2510: recipe for target 'src/CMakeFiles/minetestserver.dir/gettext.cpp.o' failed
make[2]: *** [src/CMakeFiles/minetestserver.dir/gettext.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:147: recipe for target 'src/CMakeFiles/minetestserver.dir/all' failed
make[1]: *** [src/CMakeFiles/minetestserver.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/src/gettext.h
+++ b/src/gettext.h
@@ -25,6 +25,16 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #if USE_GETTEXT
 	#include <libintl.h>
 #else
+	// In certain environments, some standard headers like <iomanip>
+	// and <locale> include libintl.h. If libintl.h is included after
+	// we define our gettext macro below, this causes a syntax error
+	// at the declaration of the gettext function in libintl.h.
+	// Fix this by including such a header before defining the macro.
+	// See issue #4446.
+	// Note that we can't include libintl.h directly since we're in
+	// the USE_GETTEXT=0 case and can't assume that gettext is installed.
+	#include <locale>
+
 	#define gettext(String) String
 #endif
 

